### ISSUE TITLE
Random fail when cloning with additional disk

### ISSUE THREAD

**Issue by sylvain324** [Telmate/terraform-provider-proxmox#876]
The goal is to create a cloud-init VM with two disks from a VM by cloning a VM with one disk.

The `main.tf` file is:

```
terraform {
  required_providers {
    proxmox = {
      source  = "telmate/proxmox"
      version = "2.9.14"
    }
  }
}


provider "proxmox" {
  pm_tls_insecure = true
  pm_log_enable   = true
  pm_log_file     = "terraform-plugin-proxmox.log"
  pm_debug        = true
  pm_log_levels   = {
    _default    = "debug"
    _capturelog = ""
  }
}


resource "proxmox_vm_qemu" "terraform-telmate-proxmox" {
  name        = "terraform-telmate-proxmox"
  target_node = "hpv-1"
  clone       = "debian-12-template"
  agent       = 1
  os_type     = "cloud-init"
  cores       = 1
  sockets     = 1
  cpu         = "host"
  memory      = 2048
  boot        = "order=virtio0"
  onboot  = true

  disk {
    type = "virtio"
    storage = "local-lvm"
    size = "15G"
  }

  disk {
    type = "virtio"
    storage = "data_disk"
    size = "20G"
  }

  network {
    model  = "virtio"
    bridge = "vmbr0"
  }

  lifecycle {
    ignore_changes = [
      network,
    ]
  }

  ipconfig0 = "ip=10.154.232.40/24,gw=10.154.232.151"
}
```

The template has three disks:
- ide0: local-lvm:vm-111-cloudinit,media=cdrom,size=4M
- ide2: none,media=cdrom
- virtio0: local-lvm:vm-111-disk-0,discard=ignore,size=14G

Sometime the provider sends the following query in the config POST request (/api2/json/nodes/hpv-1/qemu/<vmid>/config):

```
agent	1
bios	seabios
boot	order=virtio0
cores	1
cpu	host
hotplug	network,disk,usb
ipconfig0	ip=10.154.232.40/24,gw=10.154.232.151
kvm	1
memory	2048
name	terraform-telmate-proxmox
net0	virtio=AE:15:92:3A:63:98,bridge=vmbr0
numa	0
onboot	1
scsihw	lsi
sockets	1
tablet	1
virtio0	local-lvm:vm-100-disk-0,size=15G
virtio1	data_disk:20
```

In this case the VM is created as expected.

Sometime the provider sends:

```
agent	1
bios	seabios
boot	order=virtio0
cores	1
cpu	host
hotplug	network,disk,usb
ipconfig0	ip=10.154.232.40/24,gw=10.154.232.151
kvm	1
memory	2048
name	terraform-telmate-proxmox
net0	virtio=FE:2B:06:46:CD:D7,bridge=vmbr0
numa	0
onboot	1
scsihw	lsi
sockets	1
tablet	1
virtio0	local-lvm:vm-100-cloudinit,size=15G
virtio1	data_disk:20
```

In this case provider fails with the following message:

```
virtio0 - cloud-init drive is already attached at 'ide0'
```

To conclude, the provider sends randomly one of those two configurations:
- `virtio0	local-lvm:vm-100-disk-0,size=15G`
- `virtio0	local-lvm:vm-100-cloudinit,size=15G`

**Comment by hestiahacker** [comment]
This appears to be related to #704 and it may be the same issue. If so, I think it has been fixed in the default branch but not yet included in an official release. If you are willing and able to compile your own version of the provider, it seems like git commit 4c0c8fc37ed3f6316ad9392a46a0cf28c4654a11 should work (later versions either don't compile, or have other show-stopping issues; see my comments on #704 for details).

**Comment by hestiahacker** [comment]
Hello there. I saw the bot comment on this ticket being stale and for once, it was actually a useful comment. This ticket had fallen off my radar.

I made a terraform file to try to accomplish what you're trying to do here with the 3.0.1-rc1 build, making my terraform mirror yours as closely as possible. The format of disks has changed with the new version, so it's not exactly the same, but it's pretty close and it worked as expected in 3.0.1-rc1.

The main terraform file  (issue876.tf):

```
resource "proxmox_vm_qemu" "server" {
  name              = var.name
  target_node       = var.target_node
  clone             = "debian-12"
  agent             = 0
  os_type           = "cloud-init"
  scsihw            = "virtio-scsi-pci"
  # New & improved disk format (3.0 and later)
  disks {
    virtio {
      virtio0 {
        disk {
          size            = 20
          cache           = "writeback"
          storage         = var.storage_backend
        }
      }
      virtio1 {
        disk {
          size            = 40
          cache           = "writeback"
          storage         = var.storage_backend
        }
      }
    }
  }
  network {
    model           = "virtio"
    bridge          = "vmbr0"
  }
  # Cloud Init Settings
  # Reference: https://pve.proxmox.com/wiki/Cloud-Init_Support
  cloudinit_cdrom_storage = var.storage_backend
  boot = "order=virtio0;ide3"
  ipconfig0 = "ip=${var.ip_address}/${var.cidr},gw=${var.gateway}"
  nameserver = var.nameservers
  sshkeys = file("${path.root}/test.pub")
}
```

provider.tf:

```
# Define the usual provider things
terraform {
  required_providers {
    proxmox = {
      source = "Telmate/proxmox"
      version = "=3.0.1-rc1"
    }
  }
  required_version = ">= 0.14"
}
```

And my vars.tf file is specific to my environment, but you can see an example here: https://github.com/Telmate/terraform-provider-proxmox/blob/3717e8450327435cbc6fd831155b74669a6df921/examples/bug_reporting/vars.tf

If you're able to test with 3.0.1-rc1 and confirm it works, we can mark this issue to be re-tested and closed when the official version 3.x version is released.

**Comment by hestiahacker** [comment]
This is fixed in the 3.0.1-rc1 build and the fix will be released when everything in #913 has been addressed.
